### PR TITLE
SNIP-18: Staking’s First Stage on Starknet

### PR DESCRIPTION
**Pull request by ArielElp**
Suggestion for the first steps of staking in the Starknet protocol

### CODE REVIEW

File: SNIPS/snip-18.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
+---
+snip: 18
+title: Staking’s First Stage on Starknet
+author: Natan Granit <natan@starkware.co>
+discussions-to:
```
Comment (leo-starkware) [inline SNIPS/snip-18.md:5]: add link to forum thread https://community.starknet.io/t/snip-18-staking-s-first-stage-on-starknet/

File: SNIPS/snip-18.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+### Minting curve
+
+The minting curve addresses two critical aspects of the staking protocol by making the rewards dependent on the amount of STRK locked in the protocol. Here's how it works:
+
+* **Balancing Participation:** The more STRK tokens locked in the staking protocol, the lower the rewards become. This design ensures that not all STRK tokens are staked, allowing the Starknet community to continue using STRK for transaction fees and other activities. Additionally, it prevents unnecessary dilution of the token supply.
+
+* **Encouraging Staking:** When fewer STRK tokens are staked, the rewards get higher. This incentivizes participation in the staking protocol, guaranteeing a minimum level of economic security for the network by ensuring that a portion of STRK tokens is always staked.
+In the first stage, the minting curve will be based on Professor Noam Nisan's [proposal](https://starkware.co/a-token-minting-proposal-to-manage-inflation/), with a slight parameter variation. The minting curve is defined by the following formula:
+
+`M = C/10 * √ S`
```
Comment (Eikix) [inline SNIPS/snip-18.md:71]: Can you give examples right below by injecting a few "scenario" value?

e.g.

C = 2.5 and S = 30%?
C = 4 and S = 50%?

by taking examples from Cosmos chains, Solana, and Eth L1?

File: SNIPS/snip-18.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+By adopting this step-by-step approach, with the proposed first stage, we can gather valuable data, perform crucial tests, and thus refine the staking protocol as we implement it.
+
+## Specification
+
+### Overview
+
+The proposed first stage protocol features permissionless staking of STRK on Starknet with two staking flavors:
+
+* **Staking:** As a Staker, you can stake any amount of STRK greater than X STRK (X is between 10K-100K). Currently, Stakers are expected to run full nodes in preparation for the following stages. In the future, they will also need to run additional software that sequences and validates Starknet blocks and possibly perform additional network liveness and security tasks.
```
Comment (gaetbout) [inline SNIPS/snip-18.md:42]:
```suggestion
* **Staking:** As a Staker, you can stake any amount 'X' of STRK (X is between 10K-100K). Currently, Stakers are expected to run full nodes in preparation for the following stages. In the future, they will also need to run additional software that sequences and validates Starknet blocks and possibly perform additional network liveness and security tasks.
```

Comment (gaetbout) [inline SNIPS/snip-18.md:42]: Is there an upper limit of how many Stakers you expect/would like to have?
Or is it going to be like on Atom where the X Stakers with most STRK will be receiving rewards?

**Comment by gaetbout** [comment]
I don't know if related, but what happens to the tokens currently delegated on the governance?
Also, do delegating tokens to a Staker allow him to vote then?

File: SNIPS/snip-18.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+* **Stake Delegation:** Delegators choose a Staker to whom they delegate their Stake, sharing rewards with said Staker. Stake Delegators do not need to run any software, as their chosen Staker runs it for them. This makes participation accessible to a broader audience, allowing users to participate in the Staking protocol without the technical and capital limitations.
+
+Both Stakers and Stake Delegators can unstake their funds, subject to protocol-defined latencies that ensure network stability and security. More details on these latencies can be found in the "Latencies" section.
+
+### Staking Rewards
+
+Staking rewards will be based on token minting. The rewards are calculated according to a minting curve, which follows Professor Noam Nisan’s [proposal](https://starkware.co/a-token-minting-proposal-to-manage-inflation/
```
Comment (byteZorvin) [inline SNIPS/snip-18.md:49]: I am not able to access this link, is this a closed document?

Comment (gaetbout) [reply SNIPS/snip-18.md:49]: I'm able to open it, even in private navigation, so maybe you had a network issue or something?

File: SNIPS/snip-18.md (hunk 3 above)
Comment (natan-granit) [reply SNIPS/snip-18.md]: We don't plan to have a limit or have the Atom mechanism. Instead, if the number of Stakers exceeds 100, we will choose a committee of 100 randomly (with the probability to be chosen proportional to the staking power) for each epoch.

**Pull request merged**